lib/collection/src/collection_manager/segments_updater.rs

```diff
@@ -50,6 +52,19 @@ pub(crate) fn delete_points(
     Ok(res)
 }
 
+/// Deletes points from all segments which match the given filter
+pub(crate) fn delete_points_by_filter(
+    segments: &SegmentHolder,
+    op_num: SeqNumberType,
+    filter: &Filter,
+) -> CollectionResult<usize> {
+    let res = segments.apply_segments(|s| {
+        let deleted = s.delete_filtered(op_num, filter)?;
+        Ok(AppliedSegmentResult::AffectedPoints(deleted.len()))
+    })?;
+    Ok(res.1)
```

Comment:
```suggestion
    let mut deleted = 0;
    let res = segments.apply_segments(|s| {
        deleted  += s.delete_filtered(op_num, filter)?;
        Ok(true)
    })?;
    Ok(deleted)
```
that is what I actually meant

Reply:
Makes total sense! I should have stumbled upon this myself. Thanks for the hint! I still don't like the design too much, but at least its clean and more straight forward right now.